src/org/zaproxy/zap/extension/params/ExtensionParams.java

```diff
@@ -349,6 +350,15 @@ public boolean onHttpResponseReceive(HttpMessage msg) {
 			persist(sps.addParam(site, iter.next(), msg));
 		}
 
+		// Header "Parameters"
+		List<HttpHeaderField> headersList = msg.getResponseHeader().getHeaders();
+		Iterator<HttpHeaderField> hdrIter = headersList.iterator();
+		while (hdrIter.hasNext()) {
+			HttpHeaderField hdrField = hdrIter.next();
+			HtmlParameter headerParam = new HtmlParameter(HtmlParameter.Type.header, hdrField.getName(), hdrField.getValue());
+			sps.addParam(site, headerParam, msg);

```

Comment:
It's not persisting the headers (`SiteParameters` would have to be changed too, though).